### PR TITLE
Move check labels to separate workflow

### LINKED ISSUE #978: [PR Label enforcer] When a label is added to a PR, trigger the worflow to re-verify the existence of the label
Currently folks have two options for getting their "missing label" ci check to pass:

1. Make a new commit
2. Manually rerun the workflow (if they even have permission to do so)

Instead, the CI should do this automatically.

This can be done in one of two ways:

1. Quick fix: Add a `on:pull_request.labeled` trigger to the lint.yml workflow
2. Proper fix: Move the label check job to it's own workflow, trigger it on pulls and `on:pull_request.labeled`, and ensure it can block CI

### PR DESCRIPTION
**Pull request by clee2000**
* moves check labels to separate workflow that is triggered on the usual pull_request triggers as well as labeled and unlabeled
* deletes comments when label is added

Fixes https://github.com/pytorch/test-infra/issues/978 and https://github.com/pytorch/pytorch/issues/87865

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
LGTM! One the other hand, I'm a bit confused between this new `check-labels` workflow and the existing `is-properly-labeled` one in https://github.com/pytorch/pytorch/blob/master/.github/workflows/pr-labels.yml.  Now that we are enforcing the label check in the PR, running `is-properly-labeled` does no harm, but a bit redundant I think

**Comment by clee2000** [comment]
> LGTM! One the other hand, I'm a bit confused between this new check-labels workflow and the existing is-properly-labeled one in https://github.com/pytorch/pytorch/blob/master/.github/workflows/pr-labels.yml. Now that we are enforcing the label check in the PR, running is-properly-labeled does no harm, but a bit redundant I think

I think now that we block on labels, that workflow can be removed.

File: .github/scripts/check_labels.py [hunk 1]
```diff
@@ -40,21 +39,15 @@ def delete_comment(comment_id: int) -> None:
...
-    is_not_user_facing_pr = any(label.strip() == "topic: not user facing" for label in pr_labels)
-    if is_not_user_facing_pr:
-        return True
-
-    # Check if bot has already posted a message within the past hour to include a release notes label
+    # Delete all previous comments
     for comment in pr.get_comments():
         if comment.body_text.lstrip(" #").startswith(ERR_MSG_TITLE) and comment.author_login in BOT_AUTHORS:
-            ts = datetime.strptime(comment.created_at, "%Y-%m-%dT%H:%M:%SZ")
-            if (datetime.utcnow() - ts) < timedelta(hours=1):
```
Comment (ZainRizvi) [inline .github/scripts/check_labels.py:52]: This old check was there to rate limit ourselves if a person is pushing multiple commits to their PR but they haven't yet added a label it it.

Perhaps it's better to split this into two functions:
- has_required_labels(pr) - which only returns a bool and doesn't delete any comments
- delete_stale_comments(pr, has_required_labels: bool) - If the labels are there, then it deletes it's old comment. If the label still isn't there, then it'll only delete the comment if it was last posted more than an hour ago (the old behavior)

**Review by ZainRizvi — CHANGES_REQUESTED** [review]

**Comment by ZainRizvi** [comment]
> I think now that we block on labels, that workflow can be removed.

+1 to removing that old one

File: .github/scripts/check_labels.py (hunk 1 above)
Comment (clee2000) [reply .github/scripts/check_labels.py]: * split up into check for labels, add comment, and delete comment
* delete comments if has labels
* add comment if does not have label and does not have comment yet - not quite the same but i dont want to spam with extra comments when one already exists

**Pull request closed without merging**